src/components/VideoPlayer/BaseVideoPlayer.js

```diff
@@ -253,13 +255,20 @@ function BaseVideoPlayer({
                                             style={[styles.w100, styles.h100, videoPlayerStyle]}
                                             videoStyle={[styles.w100, styles.h100, videoStyle]}
                                             source={{
-                                                uri: sourceURL,
+                                                // if video is loading and is offline, we want to change uri to null to
+                                                // reset the video player after connection is back
+                                                uri: !isLoading || (isLoading && !isOffline) ? sourceURL : null,
                                             }}
                                             shouldPlay={false}
                                             useNativeControls={false}
                                             resizeMode={resizeMode}
                                             isLooping={isLooping}
-                                            onReadyForDisplay={onVideoLoaded}
```

Comment:
✋ Coming from https://github.com/Expensify/App/issues/55965
According to the [Expo documentation](https://docs.expo.dev/versions/latest/sdk/video-av/#onreadyfordisplay:~:text=A%20function%20to%20be%20called%20when%20the%20video%20is%20ready%20for%20display.%20Note%20that%20this%20function%20gets%20called%20whenever%20the%20video%27s%20natural%20size%20changes.) on `onReadyForDisplay`:
> A function to be called when the video is ready for display. Note that this function gets called whenever the video's natural size changes.

This means the video was starting playing every time you toggle full screen. This was resolved by using `onLoad` instead